transmute_core/tests/test_schematics.py

```diff
@@ -165,3 +165,27 @@ def test_can_handle(object_serializer_set, cls, should_handle):
     the object serializer set should not raise an exception if it can handle these types.
     """
     object_serializer_set[cls]
+
+
+def test_schematics_uses_cached_entries():
+    """
+    Regression test. Validating that SchematicsSerializer uses the
+    same json schema it used previously, as before erroneous
+    use of the instantiated model as a key was causing memory leaks.
+    """
+    serializer = SchematicsSerializer()
+    # A nested schema type is required as primitives have
+    # hard-coded dictionaries representing the json.
+    class SchematicsBody(Model):
+        name = StringType(max_length=5)
+
+    # ensure that instances have the same key as well.
+    instance = ModelType(SchematicsBody)
+    original_payload = serializer.to_json_schema(instance)
+    second_payload = serializer.to_json_schema(instance)
+    assert original_payload is second_payload
+
+    # classes are also valid output to recieve that the json schema
+    original_payload = serializer.to_json_schema(SchematicsBody)
```

Comment:
These `SchematicsBody` classes do not get instantiated within the `to_json_schema` method because it fails the `issubclass(model_or_class, BaseType)` check in `_enforce_instance`.

Reply:
SchematicsBody classes should not get instantiated. you could argue _enforce_instance is a misnomer here, as schematics has an incongruent apis between "types" and "models":

* types need to be an instance to call validate
* models need to be a class to instantiate and call validate

that incongruency leads to a lot of the code that exists in this file, dealing with specific edge cases around the various classes and their similar, but slightly mismatched, APIs. 

to clarify, I could rename this method to "_enforce_instance_type_or_model". How does that sound?

Reply:
Ahh, got it, thanks for clarifying. That change sounds great!